### PR TITLE
feat: added typings for vue-analytics

### LINKED ISSUE #161: TypeScript support
Hello, thanks for your work. Could you add TypeScript support by adding [declaration file](https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html)?

### PR DESCRIPTION
**Pull request by justindra**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Added Typings for Vue Analytics, closes #161 

**What is the current behavior? (You can also link to an open issue here)**
There is no TS support #161 

**What is the new behavior (if this is a feature change)?**
TS Support added

**Does this PR introduce a breaking change?**
Nope

**Please check if the PR fulfills these requirements**
- [x] The commit message follows semantic-release [guidelines](https://github.com/semantic-release/semantic-release#commit-message-format)
- [x] Fix/Feature: Docs have been added/updated
- [x] Fix/Feature: Tests have been added; existing tests pass

**Other information**:
N/A

### CODE REVIEW

**Comment by MatteoGabriele** [comment]
@justindra thanks a lot fo the PR! I will have a look later today

File: vue-analytics.d.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
+
+declare module 'vue-analytics' {
+  import _Vue, { PluginFunction } from 'vue';
+  import VueRouter, { Route } from 'vue-router';
+
+  interface eventFn {
+    (category: string, action?: string, label?: string, value?: number): void;
+    (options: {
+      eventCategory: string,
+      eventAction: string,
```
Comment (johnjcamilleri) [inline vue-analytics.d.ts]: Lines 10-12, these should also be marked as optional, i.e.:
```
eventAction?: string,
eventLabel?: string,
eventValue?: number
```

Comment (justindra) [reply vue-analytics.d.ts]: @johnjcamilleri fixed in #260

**Comment by romansp** [comment]
Thanks for the typescript support! Any chance `options` argument of `install` method can be extracted into its own interface for proper type-checking in `Vue.install` call?

```
import VueAnalitycs, { InstallOptions } from "vue-analytics";

// type-checking here
const options: InstallOptions = { 
  id: "xxx",
  router,
};

Vue.use(VueAnalytics, options);
```

**Comment by MatteoGabriele** [comment]
@romansp to be honest, I'm glad that a PR for it has been open because I'm not really a typescript guy so I don't know too much.
if you want to support it, be my guest and open a PR to: open source is nice especially for this :)

**Pull request merged**